services/web/server/tests/integration/projects/test_access_to_studies.py

```diff
@@ -0,0 +1,244 @@
+from simcore_service_webserver.users import setup_users
+from utils_assert import assert_status
+from utils_login import LoggedUser, UserRole
+from utils_projects import NewProject
+
+# Selection of core and tool services started in this swarm fixture (integration)
+core_services = [
+    'apihub',
+    'postgres'
+]
+
+tool_services = [
+    'adminer'
+]
+
+
+STUDY_UUID = "template-uuid-THIS_IS_A_FAKE_STUDY_UUID"
+simcore_service_webserver.studies_access.ALLOWED_TEMPLATE_IDS = [STUDY_UUID, ]
+
+# TODO: create a mock instead!!!
+# @pytest.fixture
+# def mock_template_studies(mocker):
+#     # patches ALLOWED_TEMPLATE_IDS
+#     import pdb; pdb.set_trace()
+#     print(simcore_service_webserver.studies_access.ALLOWED_TEMPLATE_IDS)
+#     with mocker.patch("simcore_service_webserver.studies_access.ALLOWED_TEMPLATE_IDS", [STUDY_UUID, ]):
+
+#         import pdb; pdb.set_trace()
+#         print(simcore_service_webserver.studies_access.ALLOWED_TEMPLATE_IDS)
+#         yield
+#     print(simcore_service_webserver.studies_access.ALLOWED_TEMPLATE_IDS)
+#     import pdb; pdb.set_trace()
+
+@pytest.fixture
+#def webserver_service(loop, docker_stack, aiohttp_server, aiohttp_unused_port, api_specs_dir, app_config):
+def webserver_service(loop, aiohttp_server, aiohttp_unused_port, api_specs_dir, app_config):
+    port = app_config["main"]["port"] = aiohttp_unused_port()
+    app_config['main']['host'] = '127.0.0.1'
+
+    app_config['storage']['enabled'] = False
+    app_config['rabbit']['enabled'] = False
+
+    app = web.Application()
+    app[APP_CONFIG_KEY] = app_config
+    setup_db(app)
+    setup_session(app)
+    setup_security(app)
+    setup_rest(app, debug=True)
+    setup_login(app)
+    setup_users(app)
+    setup_projects(app,
+        enable_fake_data=False, # no fake data
+        disable_login=False
+    )
+    setup_studies_access(app)
+
+    yield loop.run_until_complete( aiohttp_server(app, port=port) )
+
+@pytest.fixture
+def client(loop, webserver_service, aiohttp_client):
+    client = loop.run_until_complete(aiohttp_client(webserver_service))
+    yield client
+
+@pytest.fixture
+def qx_client_outdir(tmpdir, mocker):
+    """  Emulates qx output at service/web/client after compiling """
+
+    basedir = tmpdir.mkdir("source-output")
+    folders = [ basedir.mkdir(folder_name) for folder_name in ('qxapp', 'resource', 'transpiled')]
+
+    index_file = Path( basedir.join("index.html") )
+    index_file.write_text(textwrap.dedent("""\
+    <!DOCTYPE html>
+    <html>
+    <body>
+        <h1>OSPARC-SIMCORE</h1>
+        <p> This is a result of qx_client_outdir fixture </p>
+    </body>
+    </html>
+    """))
+
+    # patch get_client_outdir
+    mocker.patch.object(simcore_service_webserver.statics, "get_client_outdir")
+    simcore_service_webserver.statics.get_client_outdir.return_value = basedir
+
+
+
+
+# TESTS --------------------------------------
+async def test_access_to_invalid_study(client):
```

Comment:
what is the difference to accessing the projects beside the name? study == project ??

Reply:
they are the same. will rename all projects into study ... at some point but not now